File: NotePad - Malathi/NotePadApp/app/src/main/java/com/trimble/notepadapp/billing/BillingFactory.java
package com.trimble.notepadapp.billing;

/**
 * Created by mbommi on 10/10/2018.
 */

public class BillingFactory  {


    public  static IBilling getInstance(String type){
        IBilling billing = null;
        if(type.equals("Subscription")){
            billing = new BillingSubscription();
        }
        else if(type.equals("Donation")) {
            billing = new BillingDonation();
        }
        else if(type.equals("Time")) {
            billing = new BillingFeatureBased();
        }
        else if(type.equals("Feature")) {
            billing = new BillingTimeBased();
        }

        return billing;
    }
}

File: Aparna/Notepad/Billing/BillingFactory.cs


namespace Notepad.Billing
{
    class BillingFactory
    {
        
        public static IBilling GetBillingProvider(BillingTypeEnum billingType)
        {
            switch (billingType)
            {
                case BillingTypeEnum.TIME:
                    return new TimeBasedBilling();
                case BillingTypeEnum.FEATURE:
                    return new FeatureBasedBilling();
                case BillingTypeEnum.DONATION:
                    return new BillingWithDonation();
                case BillingTypeEnum.SUBSCRIPTION:
                    return new BillingWithSubscription();
                default: return null;
            }
        }
    }
}
